test/test_common/utility.cc

```diff
@@ -207,7 +207,23 @@ AssertionResult TestUtility::waitForCounterGe(Stats::Store& store, const std::st
   while (findCounter(store, name) == nullptr || findCounter(store, name)->value() < value) {
     time_system.advanceTimeWait(std::chrono::milliseconds(10));
     if (timeout != std::chrono::milliseconds::zero() && !bound.withinBound()) {
-      return AssertionFailure() << fmt::format("timed out waiting for {} to be {}", name, value);
+      return AssertionFailure() << fmt::format("timed out waiting for {} to be ge than {}", name,
+                                               value);
+    }
+  }
+  return AssertionSuccess();
+}
+
+AssertionResult TestUtility::waitForCounterDestroyedOrLe(Stats::Store& store,
+                                                         const std::string& name, uint64_t value,
+                                                         Event::TestTimeSystem& time_system,
+                                                         std::chrono::milliseconds timeout) {
+  Event::TestTimeSystem::RealTimeBound bound(timeout);
+  while (findCounter(store, name) != nullptr && findCounter(store, name)->value() > value) {
```

Comment:
This is not actually waiting for the counter to be destroyed.
There's an implicit assumption here that the counter was in the store before, however, if we run `waitForCounterDestroyedOrLe(store, "never_existed_counter", ...)` it will return true.

Moreover, IIUC, the server (and the counter) could potentially disappear between the left and right parts of this line, which will not prevent the ASAN failure you are trying to solve.

Reply:
Thanks. I admit there is semantic change. I confirmed that counter shared_ptr is latched after the cluster is removed. 

I restoredd in a987bbc